### DISCUSSION TITLE
Setting up a Crouton desktop under new conditions

### DISCUSSION THREAD

**Discussion by CroutonIsFun** [dnschneid/crouton#5013] [General]
_What follows is a delightfully  sneaky solution to a seemingly insurmountable problem: You can't sudo in the crosh shell, you can sudo in VT-2; however, VT-2 does not support an X session_

**Bare bones: Set up crouton in VT-2, run sshd, then from crosh shell ssh into localhost and startup your desktop!**

For some of you, the above will suffice. Here is a more detailed explanation:

In the crosh shell, sudo is now a privileged command, and will not work. So, to set up crouton, you have to use VT-2, which is accessed with Ctrl-Alt-F2 (right arrow). Once there, login as chronos. No password is needed unless you previously set one.

If you have chroots which did not get setup correctly, now is a good time to clean them up with

`sudo delete-chroot foo`

Note: foo is a placeholder for the chroot name.

Note: There have been reports of sudo not working in VT-2. Apparently, this was unintentional, and is in the process of being fixed.

Now, you can start from scratch with the following command:

`sudo CROUTON_BRANCH=silence crouton -r xenial -t xfce`

However, I recommend that you use Debian bullseye or Ubuntu focal: in my experience, they both work well.

Once the chroot is installed, you will enter your username and password. Enter the command `sudo enter-chroot` and continue as below.

If you need direct access to your cdrom or sound card, use the following commands to add your username to the appropriate entry in /etc/group:

`sudo adduser foo cdrom` and
`sudo adduser foo hwaudio`

As always, foo is a placeholder; in this case, it represents your username.

Now, while you're still in the chroot, install ssh:

`sudo apt install ssh`

ssh will generate a key. 

At this point, you need to exit your chroot, and the re-enter with

`sudo enter-chroot`

Final step here in VT-2 is to activate ssh with this command:

`sudo /etc/init.d/ssh/start`

Note: you can automate the above by creating the following file as /etc/rc.local:

    #!/bin/sh -e
    #
    /etc/init.d/ssh start &
    exit 0

For example, after `sudo nano /etc/rc.local` you would create the above file, then after saving it as you exit the editor, you need to make it executable with
`sudo chmod 755 /etc/rc.local`

Now exit your chroot and jump back in with `sudo enter-chroot`

Stay inside your chroot, and get back to ChromeOS by pressing Ctrl-Alt-F1 (left arrow).

Now press Ctrl-Alt-T to get into crosh, then type `shell` and press ENTER.

To fire up your Xsession, first enter this command to get into your chroot:

`ssh foo@localhost` and press ENTER

Note: foo is a placeholder for your username.

Now, enter the command `startxfce4` and press ENTER. The desktop should launch, and you can do first time stuff, etc.

To exit your chroot, first click on the exit command from your desktop menu, then press Ctrl-D to get completely out.

If you feel safe, you can leave your chroot running in VT-2; however, you can certainly go back there and Ctrl-D from the chroot, then Ctrl-D from VT-2.

**Comment by dnschneid** [comment]
Thanks for the writeup!  A couple corrections:

> If you have chroots which did not get setup correctly, now is a good time to clean them up with
> 
> `sudo rm -R /mnt/stateful_partition/crouton/chroots/*`
> 
It's better to use `delete-chroot` to clean up chroots; there's some safeties in there to prevent things from going nuts if you happend to have something mounted.

> Then go, `sudo nano /etc/group` and add your username to cdrom. Also, if you're planning on accessing your sound card, add your username to the hwaudio group.

You shouldn't need to edit /etc/group directly.  Instead, `sudo adduser username group` will add your user to the specified group.

> `ssh -X foo@localhost` and press ENTER

`-X` is not needed (but harmless), since you don't have an X server running in Chromium OS.

You can automate some stuff; adding the ssh start command (without sudo) to your /etc/rc.local will ensure it launches when you start your chroot.  Running something like `sudo enter-chroot -b sleep inf` in vt-2 will run the /etc/rc.local (if the chroot isn't launched already) then fork into the background indefinitely.  You can also play with profiles in the SecureShell app to make shortcuts to the various commands you run to start your graphical session.

**Reply by CroutonIsFun** [reply]
Thank you kindly! I have edited my how-to to reflect these corrections.

**Comment by nebcanuck** [comment]
This is very helpful, thank you! I'll give it a try in the near future.

**Reply by CroutonIsFun** [reply]
@nebcanuck 

You are very welcome. @dnschneid gave out the idea kernel, and I ran with it. :)

**Comment by CroutonIsFun** [comment]
@dnschneid 

A few things I've noticed:

1. F̶o̶r̶k̶i̶n̶g̶ ̶t̶h̶e̶ ̶c̶h̶r̶o̶o̶t̶ ̶i̶n̶t̶o̶ ̶t̶h̶e̶ ̶b̶a̶c̶k̶g̶r̶o̶u̶n̶d̶ ̶w̶i̶t̶h̶ ̶`̶s̶u̶d̶o̶ ̶e̶n̶t̶e̶r̶-̶c̶h̶r̶o̶o̶t̶ ̶-̶b̶ ̶i̶n̶f̶`̶ ̶d̶o̶e̶s̶n̶'̶t̶ ̶w̶o̶r̶k̶ ̶a̶s̶ ̶e̶x̶p̶e̶c̶t̶e̶d̶,̶ ̶a̶s̶ ̶n̶o̶w̶ ̶m̶y̶ ̶s̶s̶h̶ ̶c̶o̶n̶n̶e̶c̶t̶i̶o̶n̶ ̶i̶s̶ ̶r̶e̶j̶e̶c̶t̶e̶d̶.̶ oops, I left out the sleep ;-) Yeah, this is useful, because I can log out of VT-2 and the chroot keeps running. Then, if I wish to end my Xsession and exit the chroot completely in the crosh shell, I can jump back in with having to restart everything from scratch. Thanks! To terminate the chroot, use `sudo unmount-chroot foo -f -y` back in VT-2.
 2.  A̶f̶t̶e̶r̶ ̶I̶ ̶s̶s̶h̶ ̶i̶n̶t̶o̶ ̶m̶y̶ ̶c̶h̶r̶o̶o̶t̶ ̶a̶n̶d̶ ̶s̶t̶a̶r̶t̶ ̶m̶y̶ ̶X̶s̶e̶s̶s̶i̶o̶n̶,̶ ̶t̶h̶e̶n̶ ̶I̶'̶m̶ ̶l̶o̶c̶k̶e̶d̶ ̶i̶n̶:̶ ̶t̶h̶e̶r̶e̶'̶s̶ ̶n̶o̶ ̶w̶a̶y̶ ̶b̶a̶c̶k̶ ̶t̶o̶ ̶C̶h̶r̶o̶m̶e̶O̶S̶ ̶b̶u̶t̶ ̶t̶o̶ ̶e̶x̶i̶t̶ ̶t̶h̶e̶ ̶d̶e̶s̶k̶t̶o̶p̶.̶ ̶P̶r̶e̶s̶s̶i̶n̶g̶ ̶C̶t̶r̶-̶A̶l̶t̶-̶S̶h̶i̶f̶t̶-̶F̶1̶ ̶o̶n̶l̶y̶ ̶t̶a̶k̶e̶s̶ ̶m̶e̶ ̶t̶o̶ ̶a̶ ̶t̶o̶p̶ ̶v̶i̶e̶w̶ ̶o̶f̶ ̶t̶h̶e̶ ̶c̶r̶o̶s̶h̶ ̶s̶h̶e̶l̶l̶,̶ ̶b̶u̶t̶ ̶n̶o̶ ̶h̶i̶g̶h̶e̶r̶.̶ ̶T̶h̶i̶s̶ ̶g̶i̶v̶e̶s̶ ̶n̶e̶w̶ ̶m̶e̶a̶n̶i̶n̶g̶ ̶t̶o̶ ̶t̶h̶e̶ ̶p̶h̶r̶a̶s̶e̶,̶ ̶"̶c̶h̶r̶o̶o̶t̶ ̶j̶a̶i̶l̶"̶.̶ ̶:̶-̶)̶ ̶I̶'̶m̶ ̶g̶o̶i̶n̶g̶ ̶t̶o̶ ̶t̶e̶l̶l̶ ̶p̶e̶o̶p̶l̶e̶ ̶t̶h̶e̶ ̶c̶r̶o̶u̶t̶o̶n̶ ̶e̶x̶t̶e̶n̶s̶i̶o̶n̶ ̶i̶s̶ ̶b̶a̶s̶i̶c̶a̶l̶l̶y̶ ̶u̶s̶e̶l̶e̶s̶s̶ ̶a̶t̶ ̶t̶h̶i̶s̶ ̶p̶o̶i̶n̶t̶,̶ ̶a̶n̶d̶ ̶n̶o̶t̶ ̶t̶o̶ ̶b̶o̶t̶h̶e̶r̶ ̶i̶n̶s̶t̶a̶l̶l̶i̶n̶g̶ ̶i̶t̶.̶Ugh, I'm not sure what I was seeing. Ctrl-Alt-Shift-F1 allows me to see the crosh shell as well as my ChromeOS desktop behind it, unless I go full-screen. But I can still see my icons which are pinned to the taskbar.

**Reply by dnschneid** [reply]
1. you can also do something like `pkill sleep` and it should terminate the enter-chroot command and go through the standard unmount process
2. Crouton extension doesn't factor into the chroot/chromeos cycling.  It's for clipboard sync and xiwi.  I'm not sure what you mean by "top view of the crosh shell, but no higher".  Is it sending you to VT2 or something?

**Reply by CroutonIsFun** [reply]
Duhhh, my bad. I've lined out my comments above. I thought I couldn't get back to chromeOS. I'm looking at the crosh shell right now, and the desktop is behind it. So, I think the crouton extension is useful after all. Gee, I don't take hallucinogens either ;-O

**Comment by CroutonIsFun** [comment]
In regards to automating crouton, here is the pertinent portion of my .bashrc in ChromeOS:

    # Put your fun stuff here.
    alias Go="sudo enter-chroot -n focal -b sleep inf"
    alias Stop="sudo unmount-chroot focal -f -y"
    alias Crouton="sudo CROUTON_BRANCH=silence crouton"
    alias Foo="ssh xxxxx@localhost"

The 1st 3 aliases are designed to be executed in VT-2, and the 4th is for the crosh shell. Then, inside my chroot, I have a .bashrc with the following command:

`alias Go="startxfce4 2>/dev/null"`

So, now, it is super easy to start and stop crouton and company :-)

Old school crackers and hackers would not approve of my using caps in commands. I do this to be sure to avoid interfering with builtin Linux commands.

**Comment by DennisLfromGA** [comment]
**FYI**: 

In my experience, if you first sign-in to your (owner) profile and then login as 'chronos' on VT-2 all of your bash environment will load, like `~/.bashrc`, `~/.bash_aliases`, etc. so you can set up your aliases in your owner's profile (chronos shell).

-**DennisLfromGA**

**Reply by CroutonIsFun** [reply]
Yeah, I agree with this! At first I was scared of VT-2, due to never really needing it. But yeah, it is integrated well with your profile. The only thing I miss is being able to copy and paste from the browser. So I copy the browser command, paste it into a text file using the Text app, then display in VT-2 with something like:

`cat ~/Downloads/foo.txt`

Then, at least I can see it, and retype it. I think there is a way to get a text string into a command, but I haven't figured it out  yet.

You know what? Screw that! It is so simple to create aliases, and copy & paste works in the crosh shell, so for example, here is the command for the MrChromebox script:

`alias MrC="cd; curl -LO mrchromebox.tech/firmware-util.sh && sudo bash firmware-util.sh" `

And yes, this works in VT-2. As you said, first sign-in to your profile, then login as chronos on VT-2. If you just created an alias, then exit the shell and jump back in. And if you were already in VT-2, just sign-out and back in.

QED --- now it's a permanent part of my profile. I always back up my .bashrc to an external drive in case I have to powerwash the Chromebook.

**Reply by DennisLfromGA** [reply]
@CroutonIsFun,

> Then, at least I can see it, and retype it. I think there is a way to get a text string into a command, but I haven't figured it out yet.

You can execute a text string in a file in the Dev Console (VT-2) like this: `$(cat ~/Downloads/foo.txt)`

-**DennisLfromGA**

**Comment by yaboiwillss** [comment]
Im still having troubles i enter `sudo /etc/init.d/ssh/start` it says command not found i have done everything that has been said but this still shows up anything that i can do?

**Reply by CroutonIsFun** [reply]
@yaboiwillss 

Sorry for the delay. After sudo enter-chroot from your VT-2, then, now that your inside your chroot, you have to install ssh:

`sudo apt install ssh`

**Reply by BlackSwordmanKirito** [reply]
@CroutonIsFun 

i have done that and its installed but for some reason the command still is not found

it is sudo /etc/init.d/ssh/start right? maybe theres a different folder ill see rq by cd'ing and listing

also ignore the username its old lmao

edit: so aparently ssh isnt a directory its green so i assume ssh is a file?

edit 2: so sudo /etc/init.d/ssh promped the usage so im getting somewhere. doing sudo /etc/init.d/ssh start does nothing

prompting for status also does nothing ill try rebooting and see if its the same behavior

edit 3: even after rebooting doing any ssh commands fail to do anything they just make a new line? are we missing a step? you said above that ssh makes credentials are those important?

edit 4: surprisingly if i try to run sudo /etc/init.d/ssh start without sudo it throws this: mkdir: cannot create directory  '/var/run/sshd': permission denied

this bellow is as far as i can seem to figure out . . . whatever is happening something isn't working here ssh is bugged.

i did create that directory (sshd) and ran it without sudo and something happened finally

this is what it says now:

/etc/init.d/ssh start
      * Starting OpenBSD Secure Shell Server sshd
Could not load host key: etc/ssh/ssh_host_rsa_key
Could not load host key: etc/ssh/ssh_host_dsa_key
Could not load host key: etc/ssh/ssh_host_ecdsa_key
Could not load host key: etc/ssh/ssh_host_ed25519_key
                                                                                                                          [ OK ]

idk what to do anymore 💀

edit 5 (last edit): i decided to check if the keys existed and they do. idk why but they just wont load and using sudo results in nothing happening . . . so weird

**Reply by BlackSwordmanKirito** [reply]
@CroutonIsFun i tried doing everything i could but nothing is working. i also learned that my chromebook is using port 22 by default on crosh and idk how to well fix that either. ik that port 22 and 2222 are off limits so yeah.

i'd love a response asap on what you think wtf is happening here. i'm pretty tech savvy but even this has me scratching my head. i have never had so many issues before . . .

edit: though for clarification i havent ever had to try to set up a ssh on a chromebook sooooo 👀

**Reply by CroutonIsFun** [reply]
Sorry, I've been away for a while. First, you must install ssh. Although most basic Linux installs give you the ssh client, you need your Chromebook to be a server. So, from inside your chroot, it's `sudo apt install ssh`

Only after this can you go, `sudo /etc/init.d/ssh start` and you should not have to create any extra directories.

Yeah, port 22 is the default. You're not able to connect to port 22 because ssh has not yet been installed. At least, this is what your evidence is telling me.

I've been a bit depressed, because it seems like everyone has given up on Crouton. I don't yet consider it to be a lost cause, but it seems a lot of folks are moving on.

**Reply by tom0921** [reply]
Thank you for this terrific tutorial! I finally got Crouton completely working and am thrilled!

**Reply by BlackSwordmanKirito** [reply]
@CroutonIsFun i did install ssh, and after i did that i tried to run yk what you told me to do.

i checked and it says i can't install something that's installed already. i created the directory because it wouldn't let ssh do it itself without sudo yet the sudo command was doing nothing in my terminal.

once again i will clarify i did EVERY step as instructed but for some reason my chromebook well failed to do anything. in fact the last part proved i had ssh installed because it finally attempted to run the ssh server after the folder was created and not using sudo. i assume whats missing is something is supposed to be in the /var/run/sshd folder?

anyways point is:
1. sudo apt install ssh - this was installed and is still installed in my chroot
2. sudo /etc/init.d/ssh start - this does work but does nothing and i mean nothing. it throws no directories instantly goes to the next line and throws no errors.
3. i tried to debugging and thats where i got. ssh is installed, the command does nothing

i assume my sudo command is failing to work properly thats all i have. just for saying what you said to do it says ssh is already on the newest version. for now im just giving up on trying to get crouton working if the ssh fails to work, as i have no other way of accessing ubuntu at the moment. i do want to have crouton back as it was one of my favorite things about my chromebook.

**Comment by CroutonIsFun** [comment]
@BlackSwordsman

I'm completely baffled by what you said. I believe you. After starting ssh, did you go over to the crosh shell and attempt to ssh into your chroot?

**Reply by Apocalypse10110** [reply]
@CroutonIsFun I did what your tutorial said. Extremely helpful. However, whenever I try to start it from chros only the top part of it loads showing a thin about 10-millimeter line. In addition, nothing works, the only way I can get out is by doing the command to restart. Please help, thank you.

**Comment by BlackSwordmanKirito** [comment]
i did and it gave me the error of not being able to start ssh because of 2
reasons, one being something i dont remember and the other is because it
was using port 22

either way until a new easier method is found i might just not have a
chroot, however if someone knows how to stop chromebooks from updating
themselves, i have a recovery image that allows me to get crouton working
as it should. ty for trying to help and trust me im not giving up so easily.

**Reply by BlackSwordmanKirito** [reply]
(i tried sending this via email which is why its seperated @CroutonIsFun )

**Reply by DennisLfromGA** [reply]
@BlackSwordmanKirito,

> however if someone knows how to stop chromebooks from updating
themselves ...

If you remove rootfs verification you can modify `/etc/lsb-release` to remove or just change:
- `CHROMEOS_AUSERVER=https://tools.google.com/service/update2`

To something nonsensical, then it won't update. 

In recent versions you can enable the flag below:

> #Show Metered Toggle
> Shows a Metered toggle in the Network settings UI for WiFI and Cellular. 
> The toggle allows users to set whether a network should be considered metered for purposes of bandwith usage (e.g. for automatic updates). – ChromeOS

Then in your SSID configuration just toggle it ON:

![Screenshot 2023-11-21 11 52 31 AM](https://github.com/dnschneid/crouton/assets/3418772/6fd2cf13-a74a-4e64-83fc-dc44113be2c0)

-**DennisLfromGA**

**Comment by mintythephone** [comment]
when i select unity and enter the the chroot for some reason the command sudo /etc/init.d/ssh start   shows no errors but nothing starts i installed ssh already

**Comment by propio1225** [comment]
Is there a way to adjust the volume to hwaudio(The chromebook)speakers? I did this `sudo adduser foo hwaudio` which got the sound working but didn't make the volume adjustable. I turned on pulseaudio with `pulseaudio --start` but that had no effect on its control, it remained at the same volume.

**Comment by mintythephone** [comment]
Guys Ive found a way to install Ubuntu 16.04 unity-desktop on crouton While also following some aspects of the guide

**Reply by ImMarcoMendez** [reply]
Can you explain us bro?

**Reply by mintythephone** [reply]
Sorry for the long wait but let me explain how I did my way: step 1 just setup crouton in vt-2. step 2:enter the command sudo CROUTON_BRANCH=silence crouton -r xenial -t unity-desktop (unity-desktop installs every application while: sudo CROUTON_BRANCH=silence crouton -r xenial -t unity just install the desktop and not the applications step 3:just wait and enter the prompts like a good person after that do: sudo enter-chroot then sudo adduser (username you entered for crouton) hwaudio then: sudo apt install openssh-server after it install do: sudo mkdir /run/sshd and sudo /usr/sbin/sshd after that exit out of vt-2 and open crosh and type: shell and press enter then do: ssh (username you entered for crouton)@localhost it will ask you if you want to allow this fingerprint enter yes and then enter your password. its now finished and you can now enter startunity and you can now mess around with the desktop and do stuff

**Reply by mintythephone** [reply]
User credits:Croutonisfun and darrell-k
thanks to both of them for finding ways to make crouton still possible in 2024

**Reply by Zeninari** [reply]
ill have to test this out soon, see how it works. ill make an update if it works for me.

**Reply by Zeninari** [reply]
i cant believe it YOU ARE LEGENDS! it did work but um still the same issue as before, i get a blank ubuntu desktop on startup. any ideas on how to fix that?

edit: i fixed it using this:
sudo apt-get update  
sudo apt-get install compizconfig-settings-manager

the ubuntu plugin wasnt enabled . . . idk how thats even possible after a fresh install but aparently it can happen!

**Reply by mintythephone** [reply]
thanks please note there will be some issues like the Ubuntu desktop being blank

**Reply by mintythephone** [reply]
I've got tired of gnome desktop until I found a way to run the Ubuntu desktop

**Reply by Zeninari** [reply]
im just glad that its working at all, as when the update happened that broke crouton i was sad, very sad.

i have my own pc now but doesn't mean i cant breathe new life into my dying chromebook  using ubuntu. the chrome partition seems to be corrupted and idk how to fix it, but ubuntu is working fine. either way TY for this guide it means so much to me minty i mean it.

the last time i tried doing it i couldn't ssh that was the main issue. even though ssh was installed properly it wasnt allowing the ssh to make a connection at all, though i'm pretty sure you can find my long comment explaining the process i took if you want more details. anyways cheers mate thanks for the help!

edit: i used my old acc (@BlackSwordmanKirito) for the last response i did talking directly with @CroutonIsFun i wanted to change to my actual name, so there you go. btw now since i pinged you crouton, i do thank you for your time in trying to help me, it was greatly appreciated.

**Reply by Zeninari** [reply]
also i see a "how to fix this is just install compizconfig and enable the unity plugin and merge" from my email though i do not see it here as a comment so i assume you deleted it @mintythephone so ill explain it anyway just in case for others if they are stuck. as i was on the blank desktop.

first you need to make sure that the terminal can even be accessed if ctrl+alt+t does bring up the terminal your golden. insert as i said one by one:
sudo apt-get update
(if there is an update do sudo apt-get upgrade)
sudo apt-get install compizconfig-settings-manager

after its installed run it via the command line using ccsm

after that scroll to desktop, click on ubuntu unity plugin and there should be a checkbox for enabling it. i ignored any errors of things not working with the plugin ill deal with them later, but for now its working normally from what i can see (other than the audio which is either muted, low, or full blast)

**Comment by dejavu1987** [comment]
Hi,
I have got to where I launch by `startxfce4` But I get a black screen with white mouse pointer that I can move around but nothing works. no key shortcuts to get to shells, etc. I can only use power button and now and then I could get out of the UI and back to the crosh terminal and see the follwing logged.
```
~$ startxfce4
/usr/bin/startxfce4: Starting X server

_XSERVTransmkdir: ERROR: euid != 0,directory /tmp/.X11-unix will not be created.

X.Org X Server 1.18.4
Release Date: 2016-07-19
X Protocol Version 11, Revision 0
Build Operating System: Linux 4.15.0-140-generic x86_64 Ubuntu
Current Operating System: Linux localhost 5.4.280-22825-ga597018ecb19 #1 SMP PREEMPT Sat, 10 Aug 2024 02:55:51 +0000 x86_64
Kernel command line: cros_secure console= loglevel=7 init=/sbin/init cros_secure drm.trace=0x106 root=/dev/dm-0 rootwait ro dm_verity.error_behavior=3 dm_verity.max_bios=-1 dm_verity.dev_wait=1 dm="1 vroot none ro 1,0 6144000 verity payload=PARTUUID=c689f363-5244-4600-8382-4ba926d34d62/PARTNROFF=1 hashtree=PARTUUID=c689f363-5244-4600-8382-4ba926d34d62/PARTNROFF=1 hashstart=6144000 alg=sha256 root_hexdigest=a35614910be0363c10a3fb736d201cb30a495d8c68b23e68f808777e8974d487 salt=e9878bef2f9c6cb82f9c6c6bc90282024941edb4803fd4a7c4d9e593621994dc" noinitrd vt.global_cursor_default=0 kern_guid=c689f363-5244-4600-8382-4ba926d34d62 cros_lsb_release_hash=dmjO1JPoDLuKZ2Vqv7JCC1q40UFQFzy/cDQ0jtK9wEg= add_efi_memmap noresume i915.modeset=1 ramoops.ecc=1 tpm_tis.force=0 intel_pmc_core.warn_on_s0ix_failures=1 i915.enable_guc=2 xdomain=0 intel_iommu=on i915.enable_psr=1
Build Date: 08 April 2021  01:52:21PM
xorg-server 2:1.18.4-0ubuntu0.12 (For technical support please see http://www.ubuntu.com/support)
Current version of pixman: 0.33.6
        Before reporting problems, check http://wiki.x.org/
        to make sure that you have the latest version.
Markers: (--) probed, (**) from config file, (==) default setting,
        (++) from command line, (!!) notice, (II) informational,
        (WW) warning, (EE) error, (NI) not implemented, (??) unknown.
(++) Log file: "/tmp/Xorg.crouton.1.log", Time: Sun Sep 15 15:06:18 2024
(==) Using system config directory "/usr/share/X11/xorg.conf.d"
Error org.freedesktop.DBus.Error.ServiceUnknown: The name org.chromium.LibCrosService was not provided by any .service files
xf86EnableIOPorts: failed to set IOPL for I/O (Operation not permitted)
Error executing command as another user: Not authorized

This incident has been reported.
i965_dri.so does not support the 0x9a78 PCI ID.
crouton: version 1-20240710145429~silence:0e818d3a
release: xenial
architecture: amd64
xmethod: xorg
targets: xfce
host: version 15964.48.0 (Official Build) stable-channel volteer
kernel: Linux localhost 5.4.254-22825-ga597018ecb19 #1 SMP PREEMPT Sat, 10 Aug 2024 02:55:51 +0000 x86_64 x86_64 x86_64 GNU/Linux
freon: yes
xrandr: Output eDP1 is not disconnected but has no modes
xrandr: Output DP1 is not disconnected but has no modes
xrandr: Output DP2 is not disconnected but has no modes
xrandr: Output DP3 is not disconnected but has no modes
xrandr: Output DP4 is not disconnected but has no modes
xrandr: Output HDMI1 is not disconnected but has no modes
Kicking xrandr again
xrandr: Output eDP1 is not disconnected but has no modes
xrandr: Output DP1 is not disconnected but has no modes
xrandr: Output DP2 is not disconnected but has no modes
xrandr: Output DP3 is not disconnected but has no modes
xrandr: Output DP4 is not disconnected but has no modes
xrandr: Output HDMI1 is not disconnected but has no modes
Kicking xrandr again
_IceTransmkdir: ERROR: euid != 0,directory /tmp/.ICE-unix will not be created.
xfce4-session: No GPG agent found

(xfce4-session:6715): xfce4-session-WARNING **: xfsm_manager_load_session: Something wrong with /home/pam/.cache/sessions/xfce4-session-localhost:1, Does it exist? Permissions issue?
/usr/bin/xbindkeys_autostart: line 24: CONF: unbound variable

(xfsettingsd:6741): xfsettingsd-WARNING **: Failed to get the _NET_NUMBER_OF_DESKTOPS property.

(xfdesktop:6737): thunar-uca-WARNING **: Failed to load `/home/pam/.config/Thunar/uca.xml': Error on line 1 char 1: Document was empty or contained only whitespace

(xfce4-session:6715): Gtk-CRITICAL **: IA__gtk_widget_set_visible: assertion 'GTK_IS_WIDGET (widget)' failed

(xfce4-session:6715): GLib-GObject-CRITICAL **: g_object_unref: assertion 'G_IS_OBJECT (object)' failed
Running exit commands...
/usr/bin/xinit: connection to X server lost

waiting for X server to shut down xscreensaver: 15:14:52: SIGHUP received: restarting...
xscreensaver: 15:14:52: running as pam/pam (1000/1000)

Hangup
xfsettingsd: Fatal IO error 11 (Resource temporarily unavailable) on X server :1.0.
xscreensaver: 15:14:53: Can't open display: :1.0
xscreensaver: 15:14:53: running as pam/pam (1000/1000)

xscreensaver: 15:14:53: Errors at startup are usually authorization problems.
              But you're not logging in as root (good!) so something
              else must be wrong.  Did you read the manual and the FAQ?

              http://www.jwz.org/xscreensaver/faq.html
              http://www.jwz.org/xscreensaver/man.html

Error org.freedesktop.DBus.Error.ServiceUnknown: The name org.chromium.LibCrosService was not provided by any .service files
(II) Server terminated successfully (0). Closing log file.
```
I tried 
```
sudo apt-get update
sudo apt-get install compizconfig-settings-manager

ccsm
```
which logged:

```
~$ ccsm
/usr/lib/python2.7/dist-packages/gtk-2.0/gtk/__init__.py:57: GtkWarning: could not open display
  warnings.warn(str(e), _gtk.Warning)
Traceback (most recent call last):
  File "/usr/bin/ccsm", line 94, in <module>
    import ccm
  File "/usr/lib/python2.7/dist-packages/ccm/__init__.py", line 1, in <module>
    from ccm.Conflicts import *
  File "/usr/lib/python2.7/dist-packages/ccm/Conflicts.py", line 26, in <module>
    from ccm.Constants import *
  File "/usr/lib/python2.7/dist-packages/ccm/Constants.py", line 30, in <module>
    CurrentScreenNum = gtk.gdk.display_get_default().get_default_screen().get_number()
AttributeError: 'NoneType' object has no attribute 'get_default_screen'
```

**Reply by dejavu1987** [reply]
So I had it working until the ssh ing into the chroot and startxfce4, but the desktop was black empty and only thing visible was the mouse pointer.
Then I started over,
last time I logged in to VT-2 using root, but this time chronos.
Did get until `enter-chroot` successfully, but running `sudo /etc/init.d/ssh start` did not run the ssh server.

there is no error logged, it just simply gives me back the prompt and nothing. I think it should have said something like sshd started or sth.
I tried installing openssh-server too but nope.
I have tried purging and re installing ssh but nothing worked.

**Reply by Zeninari** [reply]
i gave up on getting crouton to work since its basicly been abandoned since version 111 (chromeos)

plus my chrome partition somehow got corrupted itself so i just decided to do a different way of doing linux, which was to fully replace the chromeos partition with any linux distro using MrChromeboxTech's tool and website.

unfortunately theres nothing i can do to help nor do i think the original author can help. i really do think crouton is dead otherwise unless the devs find another way to create the chroots without relying on the tools and other things from before they were deleted from chromeos.

but there is one thing i can say which is if you want it to work as intended, you need to rollback the chromeos version to a version before 111 using a recovery USB drive with an official backed up version. just make sure chromeos doesn't try to update itself.

i do hope whatever you decide that it works for you. have a good day @dejavu1987 

edit: and if your wondering yes i lost my chroot because when i fixed it chromeos updated and i lost the ability to open my chroot as it became currupted T-T
